TiFlash panic during br restore

Br ingests sst files of write cf into tikv without default cf, tiflash paniced while pre-decode region snapshot data.
`
[2021/03/29 21:52:00.851 +08:00] [ERROR] [<unknown>] ["DB::RawCppPtr DB::PreHandleSnapshot(DB::EngineStoreServerWrap*, DB::BaseBuffView, uint64_t, DB::SSTViewVec, uint64_t, uint64_t): Code: 49, e.displayText() = DB::Exception: Raw TiDB PK: 8000000004943341, Prewrite ts: 420446641586176000 can not found in default cf for key: 7480000000000000FF755F728000000004FF9433410000000000FAFA2A4601D13FFFFF, e.what() = DB::Exception, Stack trace:\n\n0. bin/tiflash/tiflash(StackTrace::StackTrace()+0x15) [0x367d665]\n1. bin/tiflash/tiflash(DB::Exception::Exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)+0x25) [0x36741f5]\n2. bin/tiflash/tiflash(DB::RegionData::readDataByWriteIt(std::_Rb_tree_const_iterator<std::pair<std::pair<DB::RawTiDBPK, unsigned long> const, std::tuple<std::shared_ptr<DB::StringObject<true> const>, std::shared_ptr<DB::StringObject<false> const>, DB::RecordKVFormat::InnerDecodedWriteCFValue> > > const&, bool) const+0x50c) [0x734860c]\n3. bin/tiflash/tiflash(DB::Region::readDataByWriteIt(std::_Rb_tree_const_iterator<std::pair<std::pair<DB::RawTiDBPK, unsigned long> const, std::tuple<std::shared_ptr<DB::StringObject<true> const>, std::shared_ptr<DB::StringObject<false> const>, DB::RecordKVFormat::InnerDecodedWriteCFValue> > > const&, bool) const+0x17) [0x73274a7]\n4. bin/tiflash/tiflash(DB::ReadRegionCommitCache(std::shared_ptr<DB::Region> const&, bool)+0x1c1) [0x731c4d1]\n5. bin/tiflash/tiflash(DB::GenRegionPreDecodeBlockData(std::shared_ptr<DB::Region> const&, DB::Context&)+0x30) [0x731cda0]\n6. bin/tiflash/tiflash(DB::KVStore::preHandleSnapshot(std::shared_ptr<DB::Region>, DB::SSTViewVec, DB::TMTContext&)+0x573) [0x76fb023]\n7. bin/tiflash/tiflash(DB::PreHandleSnapshot(DB::EngineStoreServerWrap*, DB::BaseBuffView, unsigned long, DB::SSTViewVec, unsigned long, unsigned long)+0xcf) [0x732678f]\n8. /tidb-deploy-upgrade0/tiflash-9900/bin/tiflash/libtiflash_proxy.so(+0x11b14cc) [0x7f8f3eb304cc]\n9. /tidb-deploy-upgrade0/tiflash-9900/bin/tiflash/libtiflash_proxy.so(+0x119a60e) [0x7f8f3eb1960e]\n10. /tidb-deploy-upgrade0/tiflash-9900/bin/tiflash/libtiflash_proxy.so(+0x9c6112) [0x7f8f3e345112]\n11. /tidb-deploy-upgrade0/tiflash-9900/bin/tiflash/libtiflash_proxy.so(+0x1c63770) [0x7f8f3f5e2770]\n12. /tidb-deploy-upgrade0/tiflash-9900/bin/tiflash/libtiflash_proxy.so(+0x1c57272) [0x7f8f3f5d6272]\n13. /tidb-deploy-upgrade0/tiflash-9900/bin/tiflash/libtiflash_proxy.so(+0x1c58c03) [0x7f8f3f5d7c03]\n14. /tidb-deploy-upgrade0/tiflash-9900/bin/tiflash/libtiflash_proxy.so(+0x1452082) [0x7f8f3edd1082]\n15. /lib64/libpthread.so.0(+0x7dd4) [0x7f8f3d144dd4]\n16. /lib64/libc.so.6(clone+0x6c) [0x7f8f3cb6beac]\n"] [thread_id=10]
`

--- a/dbms/src/Storages/Transaction/PartitionStreams.cpp
+++ b/dbms/src/Storages/Transaction/PartitionStreams.cpp
@@ -367,10 +367,21 @@ RegionTable::ResolveLocksAndWriteRegionRes RegionTable::resolveLocksAndWriteRegi
 /// pre-decode region data into block cache and remove
 RegionPtrWrap::CachePtr GenRegionPreDecodeBlockData(const RegionPtr & region, Context & context)
 {
-    auto data_list_read = ReadRegionCommitCache(region);
-
-    if (!data_list_read)
-        return nullptr;
+    std::optional<RegionDataReadInfoList> data_list_read = std::nullopt;
+    try
+    {
+        data_list_read = ReadRegionCommitCache(region);
+        if (!data_list_read)
+            return nullptr;
+    }
+    catch (const Exception & e)
+    {
+        // br or lighting may write illegal data into tikv, skip pre-decode and ingest sst later.
+        LOG_WARNING(&Logger::get(__PRETTY_FUNCTION__),
+            ". Got error while reading region committed cache: " << e.displayText() << ". Skip pre-decode and keep original cache.");
+        // set data_list_read and let apply snapshot process use empty block
+        data_list_read = RegionDataReadInfoList();
+    }
 
     auto metrics = context.getTiFlashMetrics();
     const auto & tmt = context.getTMTContext();

